### ISSUE TITLE
Letsencrypt doesn't seem to work. Auto generation not done.

### ISSUE THREAD

**Issue by geoffreycrofte** [WordOps/WordOps#208]
Hello there,

I did a migration from Easy Engine to WordOps and things are going quit well.
But time ran and the autogeneration of letsencrypt certificate seems to have failed.

I tried to do it manually by using the command:
`/etc/letsencrypt/acme.sh --renew-all --ecc`

Then I found another acme.sh lost in /opt/acme.sh/acme.sh so I launch the same command.
But both end up with nothing. Like everything went well (with the little green arrow of ohMyZsh telling me it's ok).

How can I force the generation of the letsencrypt certificate?
How can I make it automatic? I thought WO was shipped with auto generation.

So, for the moment, my 4 websites are down due to no valide certificate :/

System Information

- [ ] `lsb_release -a`

```
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 8.11 (jessie)
Release:	8.11
Codename:	jessie
```

- [ ] `wo info`

```
NGINX (1.16.1 ):

user                             www-data
worker_processes                 auto
worker_connections               50000
keepalive_timeout                8
fastcgi_read_timeout             300
client_max_body_size             100m
allow                            127.0.0.1 

PHP (7.2.22-1):

user                             
expose_php                       Off
memory_limit                     128M
post_max_size                    100M
upload_max_filesize              100M
max_execution_time               300

Information about www.conf
ping.path                        /ping
pm.status_path                   /status
process_manager                  ondemand
pm.max_requests                  1500
pm.max_children                  50
pm.start_servers                 10
pm.min_spare_servers             5
pm.max_spare_servers             15
request_terminate_timeout        300
xdebug.profiler_enable_trigger   off
listen                           php72-fpm.sock

Information about debug.conf
ping.path                        /ping
pm.status_path                   /status
process_manager                  ondemand
pm.max_requests                  1500
pm.max_children                  50
pm.start_servers                 10
pm.min_spare_servers             5
pm.max_spare_servers             15
request_terminate_timeout        300
xdebug.profiler_enable_trigger   on
listen                           127.0.0.1:9172

PHP (7.3.9-1):

user                             
expose_php                       Off
memory_limit                     128M
post_max_size                    100M
upload_max_filesize              100M
max_execution_time               300

Information about www.conf
ping.path                        /ping
pm.status_path                   /status
process_manager                  ondemand
pm.max_requests                  1500
pm.max_children                  50
pm.start_servers                 10
pm.min_spare_servers             5
pm.max_spare_servers             15
request_terminate_timeout        300
xdebug.profiler_enable_trigger   off
listen                           php73-fpm.sock

Information about debug.conf
ping.path                        /ping
pm.status_path                   /status
process_manager                  ondemand
pm.max_requests                  1500
pm.max_children                  50
pm.start_servers                 10
pm.min_spare_servers             5
pm.max_spare_servers             15
request_terminate_timeout        300
xdebug.profiler_enable_trigger   on
listen                           127.0.0.1:9173

MySQL (10.1.41-MariaDB) on localhost:

port                             3306
wait_timeout                     600
interactive_timeout              28800
max_used_connections             1
datadir                          /var/lib/mysql/
socket                           /var/run/mysqld/mysqld.sock
my.cnf [PATH]                    /etc/mysql/conf.d/my.cnf
```

- [ ] `nginx -V`

```
nginx version: nginx/1.16.1 (WordOps Nginx-wo)
built with OpenSSL 1.1.1c  28 May 2019
TLS SNI support enabled
configure arguments: --with-cc-opt='-g -O2 -fstack-protector-strong -Wformat -Werror=format-security -fPIC -D_FORTIFY_SOURCE=2' --with-ld-opt='-Wl,-z,relro -fPIC' --prefix=/usr/share/nginx --conf-path=/etc/nginx/nginx.conf --http-log-path=/var/log/nginx/access.log --error-log-path=/var/log/nginx/error.log --lock-path=/var/lock/nginx.lock --pid-path=/run/nginx.pid --http-client-body-temp-path=/var/lib/nginx/body --http-fastcgi-temp-path=/var/lib/nginx/fastcgi --http-proxy-temp-path=/var/lib/nginx/proxy --http-scgi-temp-path=/var/lib/nginx/scgi --http-uwsgi-temp-path=/var/lib/nginx/uwsgi --build='WordOps Nginx-wo' --with-debug --with-pcre-jit --with-file-aio --with-threads --with-http_ssl_module --with-http_v2_module --with-http_stub_status_module --with-http_realip_module --with-http_v2_hpack_enc --with-http_auth_request_module --with-openssl-opt='enable-ec_nistp_64_gcc_128 enable-tls1_3 no-ssl3-method' --with-http_addition_module --with-http_geoip_module --with-http_gzip_static_module --with-http_mp4_module --with-http_sub_module --with-http_xslt_module --add-module=/usr/src/packages/BUILD/debian/modules/headers-more-nginx-module --add-module=/usr/src/packages/BUILD/debian/modules/nginx-auth-pam --add-module=/usr/src/packages/BUILD/debian/modules/nginx-cache-purge --add-module=/usr/src/packages/BUILD/debian/modules/nginx-echo --add-module=/usr/src/packages/BUILD/debian/modules/nginx-upstream-fair --add-module=/usr/src/packages/BUILD/debian/modules/ngx_http_substitutions_filter_module --add-module=/usr/src/packages/BUILD/debian/modules/memc-nginx-module --add-module=/usr/src/packages/BUILD/debian/modules/srcache-nginx-module --add-module=/usr/src/packages/BUILD/debian/modules/HttpRedisModule --add-module=/usr/src/packages/BUILD/debian/modules/redis2-nginx-module --add-module=/usr/src/packages/BUILD/debian/modules/ngx_devel_kit --add-module=/usr/src/packages/BUILD/debian/modules/set-misc-nginx-module --add-module=/usr/src/packages/BUILD/debian/modules/nginx-module-vts --add-module=/usr/src/packages/BUILD/debian/modules/ngx_brotli
```

- [ ] `wo -v`

```
WordOps v3.9.8.12
Copyright (c) 2019 WordOps.
```

- [ ] `wp cli info --allow-root`

```
OS:	Linux 3.16.0-4-amd64 #1 SMP Debian 3.16.43-2+deb8u5 (2017-09-19) x86_64
Shell:	/usr/bin/zsh
PHP binary:	/usr/bin/php7.3
PHP version:	7.3.9-1+0~20190902.44+debian8~1.gbpf8534c
php.ini used:	/etc/php/7.3/cli/php.ini
WP-CLI root dir:	phar://wp-cli.phar/vendor/wp-cli/wp-cli
WP-CLI vendor dir:	phar://wp-cli.phar/vendor
WP_CLI phar path:	/home/geoffrey
WP-CLI packages dir:	
WP-CLI global config:	
WP-CLI project config:	
WP-CLI version:	2.4.0
```

- [ ] `curl -sL git.io/fjAp3 | sudo -E bash -`
[wordops.txt](https://transfer.vtbox.net/bammI/wordops.txt)

Thaaaaaaanks

**Comment by VirtuBox** [comment]
Hello @geoffreycrofte,
At first, please update WordOps with `wo update`, as current release is v3.10.3.

Then cleanup existing certificates with `wo site update site.tld -le=clean`, and try to issue new certificates with `wo site update site.tld -le`

**Comment by geoffreycrofte** [comment]
Hello @VirtuBox 

Thanks for the quick answer.
The update went well :)

`wo site update flexbox.ninja -le=clean`
leads to the error:
`Unable to Search string flexbox.ninja in /var/www/22222/conf/nginx/ssl.conf`

I checked /var/www/22222/ but there is no conf folder there, only `cert`, `htdocs` and `logs`.
I tried 
`wo site update flexbox.ninja -le` just in case, but:
`SSl is already configured for given site`

That doesn't make any sense to me. 
Like the certificate are else where but WO knows there is certificates but can't deal with those.

Thanks a lot :)

**Comment by VirtuBox** [comment]
Ah ok, this is due to the migration from `letsencrypt-auto` to `acme.sh` to handle SSL certificates. 

Try to run `wo stack upgrade` to deploy new configurations.
Then try again `wo site update site.tld -le=clean` or `wo site update site.tld -le=off` before issuing a new cert with `wo site update site.tld -le`.

**Comment by geoffreycrofte** [comment]
That's what I was wondering but couldn't explain it as well as you did :D 

I did a stack upgrade then I tried
`wo site update flexbox.ninja -le=clean`

It gave me

```
Testing Nginx configuration     [OK]
Testing Nginx configuration     [OK]
service nginx reload failed. check issues with `nginx -t` command
```

So I did: `nginx -t`

```
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/var/www/22222/cert/22222.crt"
nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
```

I tried then: `wo site update flexbox.ninja -le=off `
But same issue

```
Testing Nginx configuration     [OK]
service nginx reload failed. check issues with `nginx -t` command
```

```
nginx -t
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/var/www/22222/cert/22222.crt"
nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
```

I'm not sure why it is trying to get stuff inside 22222?

Thanks for the help @VirtuBox <3

**Comment by VirtuBox** [comment]
Hello @geoffreycrofte, this is a normal behavior because WordOps generate a self-signed certificate to secure the backend on port 22222 and ssl_stapling cannot be used with self-signed certs. 
But are you able to reload nginx manually with `sudo service nginx reload` ?

**Comment by geoffreycrofte** [comment]
Oh… no I can't.
We already solved this together last time on another issue.
I don't get it since I didn't touch my server since last time.

So when I try `sudo service nginx reload` I get:

`Job for nginx.service failed. See 'systemctl status nginx.service' and 'journalctl -xn' for details.`

Then `systemctl status nginx.service` i get:

```
nginx.service - A high performance web server and a reverse proxy server
   Loaded: loaded (/lib/systemd/system/nginx.service; disabled)
  Drop-In: /etc/systemd/system/nginx.service.d
           └─limits.conf
   Active: failed (Result: exit-code) since Sun 2019-11-24 20:07:04 CET; 24h ago
  Process: 792 ExecStart=/usr/sbin/nginx (code=exited, status=1/FAILURE)
  Process: 791 ExecStartPre=/usr/sbin/nginx -t (code=exited, status=0/SUCCESS)

Nov 25 21:01:14 vps297549.ovh.net systemd[1]: Unit nginx.service cannot be reloaded because it is inactive.
Nov 25 21:05:15 vps297549.ovh.net systemd[1]: Unit nginx.service cannot be reloaded because it is inactive.
Warning: Journal has been rotated since unit was started. Log output is incomplete or unavailable.
```

and `journal ctl -xn` returns:

```
Nov 25 21:06:18 vps297549.ovh.net postfix/cleanup[29997]: 4F30A13800E: message-id=<20191125200618.4F30A13800E@vps297549.ovh.net>
Nov 25 21:06:18 vps297549.ovh.net postfix/cleanup[29997]: warning: mysql:/etc/postfix/mysql/virtual_alias_maps.cf is unavailable. unsupported dictionary type: mysql
Nov 25 21:06:18 vps297549.ovh.net postfix/cleanup[29997]: warning: mysql:/etc/postfix/mysql/virtual_alias_maps.cf lookup error for "********@domain.tld"
Nov 25 21:06:18 vps297549.ovh.net postfix/cleanup[29997]: warning: 4F30A13800E: virtual_alias_maps map lookup problem for ********@domain.tld -- message not accepted, try again later
Nov 25 21:06:19 vps297549.ovh.net postfix/pickup[764]: warning: 4FC8313800E: message has been queued for 73 days
Nov 25 21:06:19 vps297549.ovh.net postfix/pickup[764]: 4FC8313800E: uid=33 from=<www-data>
Nov 25 21:06:19 vps297549.ovh.net postfix/cleanup[29998]: 4FC8313800E: message-id=<28ae8be745c5f83420d0e41491950b59@domain.tld>
Nov 25 21:06:19 vps297549.ovh.net postfix/cleanup[29998]: warning: mysql:/etc/postfix/mysql/virtual_alias_maps.cf is unavailable. unsupported dictionary type: mysql
Nov 25 21:06:19 vps297549.ovh.net postfix/cleanup[29998]: warning: mysql:/etc/postfix/mysql/virtual_alias_maps.cf lookup error for "********@domain.tld"
Nov 25 21:06:19 vps297549.ovh.net postfix/cleanup[29998]: warning: 4FC8313800E: virtual_alias_maps map lookup problem for ********@domain.tld -- message not accepted, try again later
```

Which I don't understand neither since I don't have any mail service on my server.
And WO don't have one, right?

thanks again @VirtuBox for your time!

**Comment by geoffreycrofte** [comment]
I removed postfix package because… useless :p 
Now the issue is that
`Unit nginx.service cannot be reloaded because it is inactive.`

I'll check with what you gave me in another issue (https://github.com/WordOps/WordOps/issues/128). But I don't know why/what something could conflict here. No more double nginx installation.

I come back to you.

**Comment by geoffreycrofte** [comment]
Ok, so: 

```
rc  libnginx-mod-http-auth-pam     1.14.1-1~dotdeb+8.1                                       amd64        PAM authentication module for Nginx
rc  libnginx-mod-http-geoip        1.14.1-1~dotdeb+8.1                                       amd64        GeoIP HTTP module for Nginx
rc  libnginx-mod-http-image-filter 1.14.1-1~dotdeb+8.1                                       amd64        HTTP image filter module for Nginx
rc  libnginx-mod-http-xslt-filter  1.14.1-1~dotdeb+8.1                                       amd64        XSLT Transformation module for Nginx
rc  libnginx-mod-mail              1.14.1-1~dotdeb+8.1                                       amd64        Mail module for Nginx
rc  libnginx-mod-stream            1.14.1-1~dotdeb+8.1                                       amd64        Stream module for Nginx
ii  nginx-common                   1.16.1-1ppa7                                              all          small, powerful, scalable web/proxy server - common files
ii  nginx-custom                   1.16.1-1ppa7                                              amd64        nginx web/proxy server with custom setup
rc  nginx-ee                       1.10.0-2                                                  all          small, powerful, scalable web/proxy server - ee files
ii  nginx-wo                       1.16.1-1ppa7                                              amd64        nginx web/proxy server with custom setup
```

Looks like nginx-ee is still here.
but `apt-get autoremove nginx*` get me `no matches found: nginx*`

`netstat -tulpn | grep 80` gave me:

```
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      556/nginx       
tcp6       0      0 :::80                   :::*                    LISTEN      556/nginx       
udp        0      0 51.254.114.191:123      0.0.0.0:*                           380/ntpd        
udp        0      0 127.0.0.1:123           0.0.0.0:*                           380/ntpd        
udp        0      0 0.0.0.0:123             0.0.0.0:*                           380/ntpd        
udp6       0      0 fe80::f816:3eff:fe9:123 :::*                                380/ntpd        
udp6       0      0 ::1:123                 :::*                                380/ntpd        
udp6       0      0 :::123                  :::*                                380/ntpd
```
I'm not really confident with what I should try to remove nginx redudancy. I'll have a look on easyengine documentation.

**Comment by geoffreycrofte** [comment]
I cleaned a lot of EE stuff here and there.
But can't find what doesn't work.
`sudo service nginx reload` give me the same ` Unit nginx.service cannot be reloaded because it is inactive.`

Sorry @VirtuBox  I'm not an expert in all of that :/

**Comment by VirtuBox** [comment]
Hello @geoffreycrofte,
try : 

```bash
# in case nginx is stopped, it cannot be reloaded
sudo service nginx restart
```

Otherwise, if nginx service is disabled, try : 

```bash
sudo systemctl enable nginx.service && sudo service nginx restart
```

**Comment by geoffreycrofte** [comment]
Yeah I did try to restart it already, but was the same issue.

I tried the enable command but something's wrong with the locales.

```
Synchronizing state for nginx.service with sysvinit using update-rc.d...
Executing /usr/sbin/update-rc.d nginx defaults
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LC_CTYPE = "UTF-8",
	LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("en_US.UTF-8").
Executing /usr/sbin/update-rc.d nginx enable
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LC_CTYPE = "UTF-8",
	LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("en_US.UTF-8").
Job for nginx.service failed. See 'systemctl status nginx.service' and 'journalctl -xn' for details.
```

I found this post about locales variable and placeholder and fallback on `etc/ssh/ssh_config` file but it did solve the issue. 

I tried a `dpkg-reconfigure locales` and chose one locale (en_US UTF-8) to test if it can solve the issue, but still the same problem.

I'm out of ideas my dear @VirtuBox  :/

**Comment by VirtuBox** [comment]
Hello @geoffreycrofte,
To update properly all your server packages, you can use `wo stack maintenance`. 
Check the file `/etc/default/locale`, it's content should be `LANG=C.UTF-8`
Then run `locale-gen`

**Comment by geoffreycrofte** [comment]
Hey,
So, I did a `wo stack upgrade`, then `wo maintenance`, then I checked the `etc/default/locale` file where the line was commented and `en_US` instead of `C` but I rewrote that part.
Then I did a `locale-gen`.

I tested `wo site update flexbox.ninja -le=clean` and `wo site update flexbox.ninja -le=off` but I get:

```
Testing Nginx configuration     [OK]
Testing Nginx configuration     [OK]
service nginx reload failed. check issues with `nginx -t` command
```

and `nginx-t` get me:

```
nginx: [warn] "ssl_stapling" ignored, issuer certificate not found for certificate "/var/www/22222/cert/22222.crt"
nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
```

The only things I found on internet are those issues : https://github.com/WordOps/WordOps/issues/58 and https://community.wordops.net/d/53-lot-of-warnings-after-nginx-t :D 

Looks like the only issue is that nginx doesn't want to reload even if the conf files are ok.
Does the `"ssl_stapling" ignored, issuer certificate not found for certificate "/var/www/22222/cert/22222.crt"` warning block nginx at some point?

Thanks @VirtuBox

**Comment by VirtuBox** [comment]
Hello @geoffreycrofte,
no it should not be an issue as WordOps is only looking for `[emerg]` message in Nginx output. 
Are you able to start/stop Nginx with `sudo service nginx stop` and `sudo service nginx start` ?

**Comment by geoffreycrofte** [comment]
Hello @VirtuBox 
It can't start and this is what `sudo systemctl status nginx.service` gave me.

```
nginx.service - A high performance web server and a reverse proxy server
   Loaded: loaded (/lib/systemd/system/nginx.service; enabled)
  Drop-In: /etc/systemd/system/nginx.service.d
           └─limits.conf
   Active: failed (Result: exit-code) since Sat 2019-11-30 18:35:54 CET; 2min 35s ago
  Process: 27399 ExecStart=/usr/sbin/nginx (code=exited, status=1/FAILURE)
  Process: 27398 ExecStartPre=/usr/sbin/nginx -t (code=exited, status=0/SUCCESS)

Nov 30 18:35:53 vps297549.ovh.net nginx[27399]: nginx: [emerg] bind() to 0.0.0.0:443 failed (98: Address already in use)
Nov 30 18:35:53 vps297549.ovh.net nginx[27399]: nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
Nov 30 18:35:53 vps297549.ovh.net nginx[27399]: nginx: [emerg] bind() to 0.0.0.0:80 failed (98: Address already in use)
Nov 30 18:35:53 vps297549.ovh.net nginx[27399]: nginx: [emerg] bind() to 0.0.0.0:22222 failed (98: Address already in use)
Nov 30 18:35:53 vps297549.ovh.net nginx[27399]: nginx: [emerg] bind() to 0.0.0.0:443 failed (98: Address already in use)
Nov 30 18:35:53 vps297549.ovh.net nginx[27399]: nginx: [emerg] bind() to [::]:80 failed (98: Address already in use)
Nov 30 18:35:54 vps297549.ovh.net nginx[27399]: nginx: [emerg] still could not bind()
Nov 30 18:35:54 vps297549.ovh.net systemd[1]: nginx.service: control process exited, code=exited status=1
Nov 30 18:35:54 vps297549.ovh.net systemd[1]: Failed to start A high performance web server and a reverse proxy server.
Nov 30 18:35:54 vps297549.ovh.net systemd[1]: Unit nginx.service entered failed state.
```

`dpkg --list | grep nginx` gave me

```
rc  libnginx-mod-http-auth-pam     1.14.1-1~dotdeb+8.1                              amd64        PAM authentication module for Nginx
rc  libnginx-mod-http-geoip        1.14.1-1~dotdeb+8.1                              amd64        GeoIP HTTP module for Nginx
rc  libnginx-mod-http-image-filter 1.14.1-1~dotdeb+8.1                              amd64        HTTP image filter module for Nginx
rc  libnginx-mod-http-xslt-filter  1.14.1-1~dotdeb+8.1                              amd64        XSLT Transformation module for Nginx
rc  libnginx-mod-mail              1.14.1-1~dotdeb+8.1                              amd64        Mail module for Nginx
rc  libnginx-mod-stream            1.14.1-1~dotdeb+8.1                              amd64        Stream module for Nginx
ii  nginx-common                   1.16.1-1ppa7                                     all          small, powerful, scalable web/proxy server - common files
ii  nginx-custom                   1.16.1-1ppa7                                     amd64        nginx web/proxy server with custom setup
rc  nginx-ee                       1.10.0-2                                         all          small, powerful, scalable web/proxy server - ee files
ii  nginx-wo                       1.16.1-1ppa7                                     amd64        nginx web/proxy server with custom setup
```


and I then tried to know what was on port :80 with `sudo netstat -tulpn | grep 80` and:

```
tcp        0      0 0.0.0.0:80              0.0.0.0:*               LISTEN      542/nginx       
tcp6       0      0 :::80                   :::*                    LISTEN      542/nginx       
udp6       0      0 fe80::f816:3eff:fe9:123 :::*                                366/ntpd
```

If I stopped Nginx, why Nginx is still on port 80 ? :D 

Thanks

**Comment by VirtuBox** [comment]
Hello @geoffreycrofte, because you have 3 different Nginx packages installed. 
At first try to remove nginx-ee and all libnginx- with : 
```bash
apt purge nginx-ee libnginx-mod-*
```
Then, if there still are nginx process running, find their PID with `ps -ef | grep nginx` and kill them with `kill -9 PID` where you have to replace PID with the process number.

**Comment by geoffreycrofte** [comment]
Thanks @VirtuBox 

I'm tired of this bug because I already cleaned that in the past thanks to your help, and it's back.

I'm working in parallel on another VPS from scratch with WordOps installed on it. Will see if it works better.
Most of the DNS resolutions are OK, waiting for the last ones to be sure I good.

I'll come back to you to close this issue if the from-scratch solution ends up quicker than my tries to solve issues on the old server.

Thanks for your time :)

**Comment by VirtuBox** [comment]
You're welcome @geoffreycrofte.
Migration on a fresh install is always the last solution, but sometimes it's also the easiest.

**Comment by geoffreycrofte** [comment]
Hello @VirtuBox 
The new installation is working like a charm and the migration wasn't that time consuming after all.

I'm sorry for the time I took to you for the issue that was maybe more related with EasyEngine thant WordOps, and thank you do much for that :)

Have a good day!
